Modules/audioop.c

```diff
@@ -259,7 +261,7 @@ st_linear2alaw(int16_t pcm_val) /* 2's complement (13-bit range) */
     if (seg >= 8)           /* out of range, return maximum value. */
         return (unsigned char) (0x7F ^ mask);
     else {
-        aval = (unsigned char) seg << SEG_SHIFT;
+        aval = (unsigned char) seg * (1 << SEG_SHIFT);
```

Comment:
Same comment here: `seg` is the value returned from `search`, and the `search` return value should always be between `0` and `size` (in this case, `8`), inclusive. So it doesn't look as though any change is needed here.

Are there code paths that somehow end up with a negative `seg` at this point?

Reply:
I reverted this change, but added an assert for non-negative seg.